esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -2051,6 +2145,20 @@ void WsWuInfo::getWorkunitThorSlaveLog(IPropertyTree* directories, const char *p
     getWorkunitThorSlaveLog(nodeGroup, ipAddress, logDate, logDir.str(), slaveNum, buf, outFile, forDownload);
 }
 
+void WsWuInfo::getWorkunitThorSlaveLogFromZAPFile(const char* thorProcess, const char* logDate, int slaveNum,
+    MemoryBuffer& buf, const char* outFile)
+{
+    StringBuffer zapWUFolder;
+    getWorkunitProcessLogPath("Thor", zapWUFolder);
+
+    VStringBuffer logFileName("%s%c%s_thorslave.%u.%s.log", zapWUFolder.str(), PATHSEPCHAR, thorProcess, slaveNum, logDate);
+    Owned<IFile> logfile = createIFile(logFileName);
+    if (!logfile)
+        throw MakeStringException(ECLWATCH_CANNOT_OPEN_FILE, "Cannot open %s.", logFileName.str());
+
+    getWorkunitLogFromZAPFile(logfile, logFileName, buf, outFile);
```

Comment:
I haven't looked at changes here and above very carefully yet,
but we do plan to have logs per workunit.
So it would better if possible, there was little or no mention of "ZAP" in related changed of code that is looking at workunits whether they are 'original' or imported.

It should if at all possible be agnostic and the code look as similar as possible.

As I say, I haven't fully looked at all the changes involved here,
and perhaps there are some cases where the workunit has references to clusters (for example) that don't exist in the environment they've been imported into, and the current code relies on them to some extent.

Let's have an offline discussion re. viewing/using imported workunits, vs existing, to explore what differences are needed..